web/api/webrpc/deals.go

```diff
@@ -0,0 +1,49 @@
+package webrpc
+
+import (
+	"context"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/curio/market"
+
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+type OpenDealInfo struct {
+	Actor        int64     `db:"sp_id"`
+	SectorNumber uint64    `db:"sector_number"`
+	PieceCID     string    `db:"piece_cid"`
+	PieceSize    uint64    `db:"piece_size"`
+	CreatedAt    time.Time `db:"created_at"`
+	SnapDeals    bool      `db:"is_snap"`
+
+	PieceSizeStr string `db:"-"`
+	CreatedAtStr string `db:"-"`
+}
+
+func (a *WebRPC) DealsPending(ctx context.Context) ([]OpenDealInfo, error) {
+	var deals []OpenDealInfo
+	err := a.deps.DB.Select(ctx, &deals, `SELECT sp_id, sector_number, piece_cid, piece_size, created_at, is_snap FROM open_sector_pieces ORDER BY created_at DESC`)
```

Comment:
what makes this a snap? How do these already have their sector numbers? If that is already figured-out, why must the user select SealNow() as that should have happened by trigger already?

Reply:
This isn't snap specific, applies to both snap and porep deal ingest.